### PR TITLE
KARAF-7099 Provide util for Configuration PIDs

### PR DESCRIPTION
**Pull request by oliverlietz**

### CODE REVIEW

**Comment by jbonofre** [comment]
It would be great to have a quick abstract / description of the PR. Thanks.

File: util/pom.xml [hunk 1]
```diff
@@ -74,6 +74,13 @@
             <artifactId>mail</artifactId>
             <scope>test</scope>
         </dependency>
+        <!-- nullability -->
+        <dependency>
+            <groupId>org.jetbrains</groupId>
+            <artifactId>annotations</artifactId>
```
Comment (jbonofre) [inline util/pom.xml:80]: Do we really need this dependency in util for null check ?

**Review by jbonofre — CHANGES_REQUESTED** [review]

File: util/pom.xml (hunk 1 above)
Comment (oliverlietz) [reply util/pom.xml:80]: In general I prefer to annotate parameters and return values but the dependency is of course not strictly required. The annotations (https://github.com/JetBrains/java-annotations) are used during analysis and compile time only and should have no negative impact.
Should I remove the annotations from ConfigurationPID?

Comment (jbonofre) [reply util/pom.xml:80]: Don't worry, I'm reworking your two PRs. Thanks !

**Comment by jbonofre** [comment]
Superseeded by #1368 1368

**Pull request closed without merging**